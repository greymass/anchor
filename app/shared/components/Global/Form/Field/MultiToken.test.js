import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import GlobalFormFieldMultiToken from './MultiToken';

const mockProps = {
  balances: { teamgreymass: { EOS: '10.0000 EOS' } },
  connection: {},
  label: 'multi token field',
  name: 'multi token field',
  settings: { account: 'teamgreymass' },
  value: '10.0000 EOS',
};

describe('GlobalFormFieldMultiToken', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<GlobalFormFieldMultiToken {...mockProps} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
