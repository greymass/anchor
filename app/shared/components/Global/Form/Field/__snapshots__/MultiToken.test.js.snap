// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalFormFieldMultiToken renders correctly 1`] = `
<div
  className="field"
>
  <label
    htmlFor="multi token field"
  >
    multi token field
  </label>
  <Input
    control={[Function]}
    defaultValue="10.0000"
    name="multi token field"
    onChange={[Function]}
    placeholder="0.0000"
    type="text"
  >
    <Dropdown
      additionLabel="Add "
      additionPosition="top"
      closeOnBlur={true}
      deburr={false}
      defaultValue="EOS"
      icon="dropdown"
      minCharacters={1}
      name="asset"
      noResultsMessage="No results found."
      onChange={[Function]}
      openOnFocus={true}
      options={Array []}
      renderLabel={[Function]}
      search={true}
      searchInput="text"
      selectOnBlur={true}
      selectOnNavigation={true}
      selection={true}
      wrapSelection={true}
    />
    <input />
  </Input>
</div>
`;
